db: Add query review tables and update QueryExecutionStatus enum

zhangvi7 opened the pull request:
## Changes
- QueryReview table for reviews
- QueryExecutionReviewer association table
- New QueryExecution enums "PENDING_REVIEW", "REJECTED", for executions in review process
- Alembic migration file

zhangvi7 committed:
db: Add query review tables and update QueryExecutionStatus enum

zhangvi7 committed:
feat: peer review backend endpoints

jczhong84 commented on querybook/server/models/query_review.py:
```diff
@@ -0,0 +1,109 @@
+        sql.Integer, sql.ForeignKey("user.id", ondelete="CASCADE"), nullable=False
+    )
+
+    status = sql.Column(
+        sql.Enum(QueryReviewStatus), nullable=False, default=QueryReviewStatus.PENDING
+    )
```
for a query review, it will link to a query execution by `query_execution_id`, which also contains the review status, it seems duplicate with this review `status` to me. maybe
1) we dont really need this `QueryReviewStatus`, just use the execution status instead
2) or for the `QueryExecutionStatus`, we only introduce a general `PENDING` status, but not specific to review pending, and no rejected status.  so every execution will start with a PENDING status.

jczhong84 commented on querybook/server/models/query_review.py:
```diff
@@ -0,0 +1,109 @@
+    status = sql.Column(
+        sql.Enum(QueryReviewStatus), nullable=False, default=QueryReviewStatus.PENDING
+    )
+    review_request_reason = sql.Column(
+        sql.String(length=description_length), nullable=True
+    )
+    rejection_reason = sql.Column(sql.String(length=description_length), nullable=True)
```
I think request and rejection reason should be both required

zhangvi7 replied:
I realized we will only have rejection reason after reviewer rejects, so will make rejection_reason  nullable=True for  review request creation

jczhong84 commented on querybook/server/logic/query_review.py:
```diff
@@ -0,0 +1,76 @@
+    )
+
+    if updated:
+        query_review.updated_at = datetime.datetime.now()
```
I think the `update()` function above already handles the `update_at` field.

zhangvi7 committed:
address comments: removed QueryReviewStatus

zhangvi7 committed:
address comments: removed QueryReviewStatus

zhangvi7 committed:
address comments

zhangvi7 commented:
Changes:
- Removed queryreviewstatus enum, use existing QueryExecutionStatus instead 
- Removed query_author_id from QueryReview table since info already exists within related query execution
- Regenerated alembic file

czgu commented on querybook/server/logic/query_review.py:
```diff
@@ -0,0 +1,74 @@
+
+@with_session
+def get_query_review_from_query_execution_id(query_execution_id: int, session=None):
+    query_execution = get_query_execution_by_id(query_execution_id, session=session)
```
why go through the route of getting query execution, we could do 
```
QueryReview.get(query_execution_id=query_execution_id, session=session)
```

zhangvi7 committed:
add adhoc page support

zhangvi7 committed:
type check for peer review params

zhangvi7 committed:
Merge pull request #20 from zhangvi7/peer-review/backend

feat: peer review backend endpoints
